### PR TITLE
Scaffold an Angular app for the new VIC appliance UI

### LINKED ISSUE #347: Scaffold an Angular app for the new VIC appliance management UI
**User Statement**
As a user I want to access a web user interface to install/manage/upgrade an existing version of the VIC appliance.

**Acceptance Criteria**
- [x] Create an Angular app with the latest Clarity UI/icon/angular installed
- [x] Set up initial routes for login, main view, getting started page and configuration
- [x] Write and plug in a mocked service that will ensure the appliance needs to be initialized first
- [x] Create a Dockerfile
- [x] Set up a Drone task to build the app and publish it

### LINKED ISSUE #348: Gather API endpoint requirements

### PR DESCRIPTION
**Pull request by jooskim**
Fixes #347 
Fixes #348 

PR acceptance checklist:

- [X] All unit tests pass
- [ ] All e2e tests pass - N/A
- [x] Unit test(s) included*
- [ ] e2e test(s) included* - N/A
- [ ] Screenshot attached and UX approved* - N/A

 *if applicable, add n/a if not

### CODE REVIEW

**Comment by jooskim** [comment]
This PR will add a new Angular app under folder `appliance-ui` which contains the following:

1. [WIP] Angular components and services for the VIC Appliance Admin UI
2. Express webserver that will host the containerized Angular app
3. Dockerfile to build the containerized web app
4. Build/publish tasks added to the Drone CI pipeline

File: .drone.yml [hunk 1]
```diff
@@ -14,9 +14,53 @@ pipeline:
   clone:
     image: plugins/git
     tags: true
-    # dont clone submodules
     recursive: false
 
+  clone-pr:
+    image: gcr.io/eminent-nation-87317/git-clone:1.1
```
Comment (andrewtchin) [inline .drone.yml]: whats this the new way to clone?

Comment (jooskim) [reply .drone.yml]: i noticed sometimes Drone doesn't take the latest commit I push so the build fails so I pinged Manjunath about this and he said this is a workaround to make sure we get the latest commit for the PR.

File: .drone.yml [hunk 2]
```diff
@@ -84,3 +128,8 @@ pipeline:
 services:
   selenium:
     image: selenium/standalone-firefox
+
```
Comment (andrewtchin) [inline .drone.yml]: i think you dont need dind

File: appliance-ui/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# usage. docker run -d -p 443:7443 \
+#                 -v ${absolute_path_to_certs_folder}:/certs
+#                 -v ${absolute_path_to_webserver_logs_folder}:/var/log/vic-appliance-ui
+#                 gcr.io/eminent-nation-87317/vic-appliance-ui:dev
+
+FROM node:8
+
+ENV TLS_CERTIFICATE=/certs/server.cert.pem
+ENV TLS_PRIVATE_KEY=/certs/server.key.pem
+ENV TLS_PORT=7443
```
Comment (andrewtchin) [inline appliance-ui/Dockerfile]: I think this should be 443 and the user can map whatever port they want

**Comment by jooskim** [comment]
`applianceConfig` is obsolete in the invision mockup. need to sync up with @lweitzman and @andrewtchin

**Pull request closed without merging**